File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
cnt = 0

def dfs(numbers, target, current, idx):
    global cnt
    if len(numbers) == idx:
        if current == target:
            cnt += 1
        return cnt
    
    dfs(numbers, target, current + numbers[idx], idx + 1)
    dfs(numbers, target, current - numbers[idx], idx + 1)


def solution(numbers, target):
    dfs(numbers, target, 0, 0)
    return cnt

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    private int cnt = 0;
    
    private void dfs(int[] numbers, int target, int current, int idx) {
        if (numbers.length == idx) {
            if (current == target) {
                cnt ++;
            }
            return;
        }
        
        dfs(numbers, target, current + numbers[idx], idx + 1);
        dfs(numbers, target, current - numbers[idx], idx + 1);
    }
    
    public int solution(int[] numbers, int target) {
        dfs(numbers, target, 0, 0);
        return cnt;
    }
}